app/next-client-app/api/datasets.ts

```diff
@@ -55,3 +88,44 @@ export async function archiveDataSets(id: number, hidden: boolean) {
     return { errorMessage: error.message };
   }
 }
+
+export async function updateDatasetDetails(
+  id: number,
+  name: string,
```

Comment:
The body should be an object passed, so we can use the type, rather than passing all of these as individual arguments.